src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonApiClient.kt

```diff
@@ -81,20 +81,20 @@ data class TemporaryAbsence(
 
 )
 
-/**
- * The response has many more fields and nested values, but only offenderNo is of interest
- */
-data class ConfirmArrivalResponse(
-
-  val offenderNo: String
-)
-
 /*
  * The response has many more fields and nested values but currently only bookingId is needed.
  */
-data class InmateDetail(val bookingId: Long)
-
-data class OffenderDetail(val offenderNo: String)
+data class InmateDetail(
+  val bookingId: Long,
+  val offenderNo: String,
+  val assignedLivingUnit: AssignedLivingUnit
```

Comment:
are we happy that this will never be optional?  `AssignedLivingUnit?`

Reply:
I checked the code in prison API and we always put the value for AssignedLivingUnit however InmateDetail do not have not null annotation. 
I think it is big job to cover null scenario here in terms of the frontend and backend so if you don't mind I will assume we never have a null here based on the code from prison-api
`            .assignedLivingUnit(AssignedLivingUnit.builder()
                .agencyId(latestBooking.getLocation().getId())
                .agencyName(LocationProcessor.formatLocation(latestBooking.getLocation().getDescription()))
                .description(latestBooking.getAssignedLivingUnit() != null ? RegExUtils.replaceFirst(latestBooking.getAssignedLivingUnit().getDescription(), "^[A-Z|a-z|0-9]+\\-", "") : null)
                .locationId(latestBooking.getAssignedLivingUnit() != null ? latestBooking.getAssignedLivingUnit().getLocationId() : null)
                .build())`

Reply:
the description can be null - I think hopefully you'll get a kotlin exception (albeit one with limited useful context) rather than return an empty/undefined value to the frontend

I'd rather it error "loudly" rather than silently show no error so this will suffice. 

(It would be nice to add more contextual info to the exception that would be thrown but as it's unexpected its not necessary)

Reply:
sorted